### PR TITLE
Futurize, py2 → py3

### PR DESCRIPTION
**Pull request by manics**
~~Started this a while ago, it's going to take a while so opening the PR in case someone else wants to contribute.~~

Add support for running `omego` under Python 2.7 and Python 3.6

There are no breaking changes to the CLI interface, but some method and class signatures have changed so this is a breaking change: `0.7.0`. It might be possible to do this in a non-breaking manner but I don't think the added complexity and potential bugs are worthwhile, especially as we don't have evidence of omego being used as a library. It's also still `<1.0.0`

The main change is to `external.py` as a result of removing the coupling between omego and the omero.py module. Previously `import omero` was run inside omego, now all interaction is though `bin/omero` which means the omego environment can be independent of the omero runtime environment. For example omego can be run under Python 3.6 whilst managing omero in a Python 2.7 environment. This is used in the travis test to test omego (Python 3.6) with the current OMERO release (Python 2.7).

### CODE REVIEW

**Comment by joshmoore** [comment]
Have now reached the point of needing a python3 compatible version of omego: https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-server/39/console

**Comment by manics** [comment]
This is ready for review, see updated description.

File: omego/external.py [hunk 1]
```diff
@@ -57,79 +113,53 @@ def set_server_dir(self, dir):
...
-
-        try:
-            # Attempt to open config.xml read-only, though this flag is not
-            # present in early versions of OMERO 5.0
-            c = self._omero.config.ConfigXml(
-                configxml, exclusive=False, read_only=True)
-        except TypeError:
-            c = self._omero.config.ConfigXml(configxml, exclusive=False)
-
+        stdout, stderr = self.omero_cli(['config', 'get'])
```
Comment (joshmoore) [inline omego/external.py]: Should this use `--show-password`?

File: test/integration/test_upgrade.py [hunk 2]
```diff
@@ -49,6 +51,17 @@ def testSkipunzip(self):
...
+        args = ["--branch=OMERO-DEV-latest"]
+        # Python 3.6 on Travis: Force OMERO to run with 2.7 instead
+        if getenv('TRAVIS_PYTHON_VERSION') == '3.6':
+            args += ['--python', 'python2.7']
+        self.upgrade(*args)
+
+    @pytest.mark.slowtest
+    @pytest.mark.skipif(
+        getenv('TRAVIS_PYTHON_VERSION') == '3.6',
+        reason='OMERO not supported on Python 3.6')
```
Comment (joshmoore) [inline test/integration/test_upgrade.py]: worth a todo somewhere for bumping this?

Comment (manics) [reply test/integration/test_upgrade.py]: https://github.com/ome/omego/issues/125

**Comment by manics** [comment]
Updated. `omero config get --show-password` was a bit more complicated to add because we still use OMERO.server 5.3.5 for testing the database upgrade process, but that doesn't support the `--show-password` flag

**Comment by manics** [comment]
A prerelease for testing sounds good!

**Comment by joshmoore** [comment]
Ok. Merging once green and tagging/releasing as `0.7.0.dev1`.

**Pull request merged**